### DISCUSSION TITLE
duckdb support

### DISCUSSION THREAD

**Discussion by mrgnw** [beekeeper-studio/beekeeper-studio#1486] [Ideas]
I'd like to add [duckdb](https://duckdb.org) support.

duckdb is similar to sqlite in that it can operate directly from a file on local storage. 
It can also run in memory and import data from .csv, .parquet, etc.
It is especially good at enabling fast analysis of large datasets locally.

Duckdb provides drivers at https://duckdb.org/docs/installation/index

This is the conversation for jetbrains adding support in their IDEs 
https://youtrack.jetbrains.com/issue/DBE-15099/Driver-for-DuckDB

I'll add more details as I explore adding this myself, but I wanted to get the conversation started.

**Comment by datamike** [comment]
duckdb is highly compatible with postgreSQL.  Is is possible that function dialectFor() in "/shared/lib/dialects/models.ts" could return 'postgresql' if the passed-in argument is 'duckdb'?
